commit 8feb499ba2c3625632210c997b49f5df515c05d4
Author: Daniel P. Berrange <berrange@redhat.com>
Date:   Tue Aug 4 18:13:09 2009 +0100

    Fix escaping of 8-bit high characters
    
    Fix  https://bugzilla.redhat.com/show_bug.cgi?id=479517
    
    * src/buf.c: Cast to 'unsigned char' before doing compare to
      avoid rejecting 8-bit high characters

diff --git a/src/buf.c b/src/buf.c
index 259175d..c802aa2 100644
--- a/src/buf.c
+++ b/src/buf.c
@@ -304,7 +304,7 @@ virBufferEscapeString(const virBufferPtr buf, const char *format, const char *st
             *out++ = 'o';
             *out++ = 's';
             *out++ = ';';
-        } else if ((*cur >= 0x20) || (*cur == '\n') || (*cur == '\t') ||
+        } else if (((unsigned char)*cur >= 0x20) || (*cur == '\n') || (*cur == '\t') ||
                    (*cur == '\r')) {
             /*
              * default case, just copy !
